### ISSUE TITLE
Avoid logging errors due to RequestCanceled errors

### ISSUE THREAD

**Issue by leth** [cortexproject/cortex#465]
There are a lot of errors logged when query requests are cancelled.
This is normal behaviour and should not be logged, as it masks genuine errors.

We should, however, ensure that these errors are included in a metric somewhere.

**Comment by leth** [comment]
We see this when people have dashboards open, refreshing at a rate faster than cortex can respond to.

**Comment by bboreham** [comment]
I see a lot of these, which I think are down to the distributor cancelling when it has (N-1) answers:

```
level=warn ts=2019-02-04T08:55:28.07467732Z caller=gokit.go:46 method=/cortex.Ingester/Query duration=97.494µs err="context canceled" msg="gRPC\n"
level=warn ts=2019-02-04T08:55:28.075459936Z caller=gokit.go:46 method=/cortex.Ingester/Query duration=89.658µs err="context canceled" msg="gRPC\n"
level=warn ts=2019-02-04T08:55:28.075643591Z caller=gokit.go:46 method=/cortex.Ingester/Query duration=70.858µs err="context canceled" msg="gRPC\n"
```

Seems like the same issue - we don't want to log "context canceled" all day. 
Nor do we want it flagged as an error in Jaeger.

**Comment by bboreham** [comment]
Closed by #1339
